Examples/test-suite/java_director_typemaps.i

```diff
@@ -0,0 +1,265 @@
+/*
+ * Test Lib/java/typemaps.i
+ */
+
+%module(directors="1") java_director
+
+%feature("director", assumeoverride=1) Quux;
+
+%include <typemaps.i>
+
+%apply bool& OUTPUT {bool& boolarg_output};
+
+%apply signed char& OUTPUT {signed char& signed_chararg_output};
+%apply unsigned char& OUTPUT {unsigned char& unsigned_chararg_output};
+
+%apply short& OUTPUT {short& shortarg_output};
+%apply unsigned short& OUTPUT {unsigned short& unsigned_shortarg_output};
+
+%apply int& OUTPUT {int& intarg_output};
+%apply unsigned int& OUTPUT {unsigned int& unsigned_intarg_output};
+
+%apply long& OUTPUT {long& longarg_output};
+%apply unsigned long& OUTPUT {unsigned long& unsigned_longarg_output};
+
+%apply long long& OUTPUT {long long& long_longarg_output};
+// %apply unsigned long long& OUTPUT {unsigned long long& unsigned_long_longarg_output};
+
+%apply float& OUTPUT {float& floatarg_output};
+%apply double& OUTPUT {double& doublearg_output};
+
+%apply bool& INOUT {bool& boolarg_inout};
+
+%apply signed char& INOUT {signed char& signed_chararg_inout};
+%apply unsigned char& INOUT {unsigned char& unsigned_chararg_inout};
+
+%apply short& INOUT {short& shortarg_inout};
+%apply unsigned short& INOUT {unsigned short& unsigned_shortarg_inout};
+
+%apply int& INOUT {int& intarg_inout};
+%apply unsigned int& INOUT {unsigned int& unsigned_intarg_inout};
+
+%apply long& INOUT {long& longarg_inout};
+%apply unsigned long& INOUT {unsigned long& unsigned_longarg_inout};
+
+%apply long long& INOUT {long long& long_longarg_inout};
+// %apply unsigned long long& INOUT {unsigned long long& unsigned_long_longarg_inout};
+
+%apply float& INOUT {float& floatarg_inout};
+%apply double& INOUT {double& doublearg_inout};
+
+%inline %{
+#include <assert.h>
+
+class Quux {
+public:
+  Quux() {}
+  virtual ~Quux() {}
+
+  virtual void director_method_bool_output(
+    bool& boolarg_output,
+
+    signed char& signed_chararg_output,
+    unsigned char& unsigned_chararg_output,
+
+    short& shortarg_output,
```

Comment:
Can you also add a test here where there is no parameter name in the wrapped virtual function and make sure that works? Something odd happened when I tried, the type seems to change.